src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -87,12 +98,24 @@ private boolean validString(Node node, String key) {
         return node != null && node.childOf(key) && Utils.isNotEmpty(Coerce.toString(node));
     }
 
+    /**
+     * Retrieve configValidationError.
+     * Validate again if the device config has changed.
+     *
+     */
+    public String getConfigValidationError() {
```

Comment:
no one is call this, so we still won't know what the issue is

Reply:
will add a log